plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ParameterResolver.kt

```diff
@@ -46,5 +47,7 @@ class InlineParameterResolver(
     override val defaultResolvedReturnTarget: ReturnTarget,
 ) : ParameterResolver {
     override fun tryResolveParameter(name: Name): ExpEmbedding? = substitutions[name]
-    override fun tryResolveReceiver(): ExpEmbedding? = substitutions[SpecialNames.THIS]
+    override fun tryResolveReceiver(isExtension: Boolean): ExpEmbedding? =
+        if (isExtension) substitutions[ExtraSpecialNames.EXTENSION_THIS]
+        else substitutions[ExtraSpecialNames.DISPATCH_THIS]
```

Comment:
Modifying the key type would probably be the more elegant way to deal with this, but it's okay to do as-is.